feat: 原理图管理支持显示子文件夹

caomei269 opened the pull request:
实现原理图资源管理中的文件夹浏览功能，包括：
1. 原理图管理支持显示子文件夹
2. 支持进入和返回文件夹操作
3. 为文件夹项添加特殊的UI显示和排序逻辑

close #965

caomei269 committed:
feat: 原理图管理页支持显示子目录

实现原理图资源管理中的文件夹浏览功能，包括：
1. 在根目录显示子目录项
2. 支持进入和返回目录操作
3. 为文件夹项添加特殊UI显示和排序逻辑

clone #965

caomei269 committed:
Update ModLocalCompFolder.vb

移除多余的检测文件数量

caomei269 commented:
应该有签名了 然后就是文件夹的图标没用弄

wyc-26 reviewed (commented):
https://github.com/PCL-Community/PCL2-CE/issues/965 还有一个读取投影文件信息没做

shimoranla commented:
应该是 close #965 而非 clone #965

caomei269 committed:
feat: 为.litematic文件添加NBT数据解析和显示功能

添加对.litematic文件的NBT数据解析，包括创建/修改时间、包围盒大小、区域数量、总方块数和总体积等信息。
在资源详情页面显示这些额外信息，提升用户对投影原理图文件的了解。

caomei269 committed:
Merge branch 'dev' of github.com:caomei269/PCL2-CE into dev

caomei269 reviewed (commented):
这个是删除多余的文件夹内文件数量检测

caomei269 commented:
> https://github.com/PCL-Community/PCL2-CE/issues/965 还有一个读取投影文件信息没做

差不多了 重审一下吧

MoYuan-CN reviewed (dismissed):
看起来没问题，需要实际测试支持

tangge233 commented on Plain Craft Launcher 2/Pages/PageVersion/PageVersionCompResource.xaml.vb:
```diff
@@ -123,12 +124,136 @@ Public Class PageVersionCompResource
+            End Select
+            
+            If Not hasFiles Then
+                MyMsgBox("该文件夹内没有文件。", "提示")
```
```suggestion
                Hint("该文件夹内没有文件。")
```

tangge233 commented on Plain Craft Launcher 2/Pages/PageVersion/PageVersionCompResource.xaml.vb:
```diff
@@ -123,12 +124,136 @@ Public Class PageVersionCompResource
+    Private Sub EnterFolderWithCheck(folderPath As String)
+        Try
+            If String.IsNullOrEmpty(folderPath) OrElse Not Directory.Exists(folderPath) Then
+                MyMsgBox("文件夹不存在或已被删除。", "错误")
```
```suggestion
                Hint("文件夹不存在或已被删除", HintType.Critical)
```

tangge233 commented on Plain Craft Launcher 2/Pages/PageVersion/PageVersionCompResource.xaml.vb:
```diff
@@ -123,12 +124,136 @@ Public Class PageVersionCompResource
+    Private Sub EnterFolder(folderPath As String)
+        Try
+            If String.IsNullOrEmpty(folderPath) OrElse Not Directory.Exists(folderPath) Then
+                MyMsgBox("文件夹不存在或已被删除。", "错误")
```
```suggestion
                Hint("文件夹不存在或已被删除", HintType.Critical)
```

tangge233 commented on Plain Craft Launcher 2/Pages/PageVersion/PageVersionCompResource.xaml.vb:
```diff
@@ -123,12 +124,136 @@ Public Class PageVersionCompResource
+            LoaderFolderRun(CompResourceListLoader, folderPath, LoaderFolderRunType.ForceRun, LoaderInput:=GetRequireLoaderData())
+        Catch ex As Exception
+            Log($"[错误] 进入文件夹失败：{ex.Message}")
+            MyMsgBox($"进入文件夹时发生错误：{ex.Message}", "错误")
```
```suggestion
            Log(ex, $"进入文件夹失败", LogLevel.Msgbox)
```

tangge233 commented on Plain Craft Launcher 2/Pages/PageVersion/PageVersionCompResource.xaml.vb:
```diff
@@ -123,12 +124,136 @@ Public Class PageVersionCompResource
+            EnterFolder(folderPath)
+        Catch ex As Exception
+            Log($"[错误] 检查文件夹失败：{ex.Message}")
+            MyMsgBox($"检查文件夹时发生错误：{ex.Message}", "错误")
```
```suggestion
            Log(ex, $"检查文件夹失败", LogLevel.Msgbox)
```

tangge233 commented on Plain Craft Launcher 2/Modules/Minecraft/ModLocalComp.vb:
```diff
@@ -924,8 +1009,107 @@ Finished:
+            If Not Directory.Exists(FolderPath) Then Return "空文件夹"
+            Return "文件夹"
+        Catch ex As Exception
+            Log($"[错误] 获取文件夹描述失败：{FolderPath}，错误：{ex.Message}", LogLevel.Debug)
```
```suggestion
            Log(ex, $"获取文件夹描述失败：{FolderPath}", LogLevel.Debug)
```

tangge233 commented on Plain Craft Launcher 2/Modules/Minecraft/ModLocalComp.vb:
```diff
@@ -961,17 +1145,53 @@ Finished:
+                            Try
+                                If LocalCompFile.IsCompFile(File.FullName, Loader.Input.CompType) Then ModFileList.Add(File)
+                            Catch ex As Exception
+                                Log($"[错误] 处理文件失败：{File.FullName}，错误：{ex.Message}", LogLevel.Debug)
```
```suggestion
                                Log(ex, $"处理文件失败：{File.FullName}", LogLevel.Debug)
```

tangge233 commented on Plain Craft Launcher 2/Modules/Minecraft/ModLocalComp.vb:
```diff
@@ -961,17 +1145,53 @@ Finished:
+                            End Try
+                        Next
+                    Catch ex As Exception
+                        Log($"[错误] 枚举文件失败：{SearchPath}，错误：{ex.Message}")
```
```suggestion
                        Log(ex, $"枚举文件失败：{SearchPath}")
```

tangge233 commented on Plain Craft Launcher 2/Modules/Minecraft/ModLocalComp.vb:
```diff
@@ -961,17 +1145,53 @@ Finished:
+                                End If
+                                If LocalCompFile.IsCompFile(File.FullName, Loader.Input.CompType) Then ModFileList.Add(File)
+                            Catch ex As Exception
+                                Log($"[错误] 处理文件失败：{File.FullName}，错误：{ex.Message}", LogLevel.Debug)
```
```suggestion
                                Log(ex, $"处理文件失败：{File.FullName}", LogLevel.Debug)
```

tangge233 commented on Plain Craft Launcher 2/Modules/Minecraft/ModLocalComp.vb:
```diff
@@ -999,6 +1219,38 @@ Finished:
+                                FolderEntry.Description = GetFolderDescription(SubDir.FullName)
+                                ModList.Add(FolderEntry)
+                            Catch ex As Exception
+                                Log($"[错误] 处理文件夹失败：{SubDir.FullName}，错误：{ex.Message}", LogLevel.Debug)
```
```suggestion
                                Log(ex, $"处理文件夹失败：{SubDir.FullName}", LogLevel.Debug)
```

tangge233 commented on Plain Craft Launcher 2/Modules/Minecraft/ModLocalComp.vb:
```diff
@@ -999,6 +1219,38 @@ Finished:
+                            End Try
+                        Next
+                    Catch ex As Exception
+                        Log($"[错误] 枚举文件夹失败：{SearchPath}，错误：{ex.Message}")
```
```suggestion
                        Log(ex, $"枚举文件夹失败：{SearchPath}")
```

tangge233 commented on Plain Craft Launcher 2/Pages/PageVersion/PageVersionCompResource.xaml.vb:
```diff
@@ -123,12 +124,136 @@ Public Class PageVersionCompResource
+                CurrentFolderPath = parentPath
+            End If
+        Catch ex As Exception
+            Log($"[错误] 路径处理失败：{ex.Message}")
```
```suggestion
            Log(ex, $"路径处理失败")
```

tangge233 commented on Plain Craft Launcher 2/Pages/PageVersion/PageVersionCompResource.xaml.vb:
```diff
@@ -188,20 +313,32 @@ Public Class PageVersionCompResource
+            Return NewItem
+        Catch ex As Exception
+            AniControlEnabled -= 1
+            Log($"[错误] 创建UI项失败：{Entry.RawFileName}，错误：{ex.Message}", LogLevel.Debug)
```
```suggestion
            Log(ex, $"创建UI项失败：{Entry.RawFileName}", LogLevel.Debug)
```

tangge233 commented on Plain Craft Launcher 2/Pages/PageVersion/PageVersionCompResource.xaml.vb:
```diff
@@ -935,12 +1117,17 @@ Install:
+                PanList.Children.RemoveAt(IndexOfUi)
+                PanList.Children.Insert(IndexOfUi, NewItem)
+            Catch ex As Exception
+                Log($"[错误] 更新UI列表项失败：{ModEntity.RawFileName}，错误：{ex.Message}", LogLevel.Debug)
```
```suggestion
                Log(ex, $"更新UI列表项失败：{ModEntity.RawFileName}", LogLevel.Hint)
```

tangge233 reviewed (changes requested):
一些小问题

caomei269 committed:
Update Plain Craft Launcher 2/Pages/PageVersion/PageVersionCompResource.xaml.vb

Co-authored-by: tangge233 <50769997+tangge233@users.noreply.github.com>

caomei269 committed:
Update Plain Craft Launcher 2/Modules/Minecraft/ModLocalComp.vb

Co-authored-by: tangge233 <50769997+tangge233@users.noreply.github.com>

caomei269 committed:
Apply suggestions from code review

Co-authored-by: tangge233 <50769997+tangge233@users.noreply.github.com>

caomei269 committed:
Apply suggestions from code review

Co-authored-by: tangge233 <50769997+tangge233@users.noreply.github.com>
